lms/djangoapps/courseware/tests/test_submitting_problems.py

```diff
@@ -577,8 +577,8 @@ def test_dropping_nochange(self):
         self.assertEqual(self.score_for_hw('homework1'), [1.0, 0.0])
         self.assertEqual(self.score_for_hw('homework2'), [1.0, 1.0])
         self.assertEqual(self.score_for_hw('homework3'), [1.0, 0.0])
-        self.assertEqual(self.earned_hw_scores(), [1.0, 2.0, 1.0])  # Order matters
-        self.check_grade_percent(0.75)
+        self.assertEqual(self.earned_hw_scores(), [])  # Order matters
+        self.check_grade_percent(0)
```

Comment:
This is the only test which unexpectedly fails because of the cache invalidation:

Without cache invalidation `grades.grade()` returns:

```
{  
   'section_breakdown':[  
      {  
         'category':u'Homework',
         'percent':0.5,
         'detail':u'Homework 1 - homework1 - 50% (1/2)',
         'label':u'HW 01'
      },
      {  
         'category':u'Homework',
         'percent':1.0,
         'detail':u'Homework 2 - homework2 - 100% (2/2)',
         'label':u'HW 02'
      },
      {  
         'category':u'Homework',
         'mark':{  
            'detail':u'The lowest 1 Homework scores are dropped.'
         },
         'percent':0.5,
         'detail':u'Homework 3 - homework3 - 50% (1/2)',
         'label':u'HW 03'
      },
      {  
         'category':u'Homework',
         'prominent':True,
         'percent':0.75,
         'detail':u'Homework Average = 75%',
         'label':u'HW Avg'
      }
   ],
   'grade':u'Pass',
   'totaled_scores':{  
      u'Homework':[  
         Score(earned=1.0,
         possible=2.0,
         graded=True,
         section=u'homework1'),
         Score(earned=2.0,
         possible=2.0,
         graded=True,
         section=u'homework2'),
         Score(earned=1.0,
         possible=2.0,
         graded=True,
         section=u'homework3')
      ]
   },
   'percent':0.75,
   'grade_breakdown':[  
      {  
         'category':u'Homework',
         'percent':0.75,
         'detail':u'Homework = 75.00% of a possible 100.00%'
      }
   ]
}
```

With cache invalidation:
```
{  
   'section_breakdown':[  
      {  
         'category':u'Homework',
         'percent':0,
         'detail':u'Homework 1 Unreleased - 0% (?/?)',
         'label':u'HW 01'
      },
      {  
         'category':u'Homework',
         'percent':0,
         'detail':u'Homework 2 Unreleased - 0% (?/?)',
         'label':u'HW 02'
      },
      {  
         'category':u'Homework',
         'mark':{  
            'detail':u'The lowest 1 Homework scores are dropped.'
         },
         'percent':0,
         'detail':u'Homework 3 Unreleased - 0% (?/?)',
         'label':u'HW 03'
      },
      {  
         'category':u'Homework',
         'prominent':True,
         'percent':0,
         'detail':u'Homework Average = 0%',
         'label':u'HW Avg'
      }
   ],
   'grade':None,
   'totaled_scores':{  
      u'Homework':[  

      ]
   },
   'percent':0.0,
   'grade_breakdown':[  
      {  
         'category':u'Homework',
         'percent':0,
         'detail':u'Homework = 0.00% of a possible 100.00%'
      }
   ]
}
```